sql/alter.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2015 The Cockroach Authors.
```

Comment:
I don't think `alter.go` is the right file to place this in. `create.go` seems appropriate, or if you feel a strong desire for separation, `create_index.go`.